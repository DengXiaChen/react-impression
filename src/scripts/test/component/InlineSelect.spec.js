import React from 'react';
import { shallow, mount } from 'enzyme';
import { InlineSelect } from '../../components/impression';

const setup = () => {
    const props = {
        defaultValue: '2',
        onChange: jest.fn(),
    };

    const temp = (
        <InlineSelect {...props}>
            <InlineSelect.Option value="1">北京</InlineSelect.Option>
            <InlineSelect.Option value="2">上海</InlineSelect.Option>
            <InlineSelect.Option value="3">南京</InlineSelect.Option>
        </InlineSelect>
    );

    const wrapper = shallow(temp);
    const wrapper1 = mount(temp);

    return {
        props,
        wrapper,
        wrapper1,
    };
};

describe('InlineSelect', () => {
    const { props, wrapper, wrapper1 } = setup();

    it('render', () => {
        expect(wrapper1.find('.inline-select').exists()).toBe(true);
        expect(wrapper1.find('.inline-select-option')).toHaveLength(3);
        expect(wrapper1.find('.inline-select-option.active').text()).toBe('上海');
    });

    it('change', () => {
        wrapper1.find('.inline-select-option').at(2).simulate('click');
        expect(props.onChange).toHaveBeenCalledWith('3', '南京', 2);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
