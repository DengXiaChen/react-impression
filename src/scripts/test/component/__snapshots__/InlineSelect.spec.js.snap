// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineSelect ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="inline-select"
    defaultValue="2"
    onChange={[Function]}
>
    <InlineSelectOption
        active={false}
        onClick={[Function]}
        value="1"
    >
        北京
    </InlineSelectOption>
    <InlineSelectOption
        active={true}
        onClick={[Function]}
        value="2"
    >
        上海
    </InlineSelectOption>
    <InlineSelectOption
        active={false}
        onClick={[Function]}
        value="3"
    >
        南京
    </InlineSelectOption>
</div>,
  "nodes": Array [
    <div
      className="inline-select"
      defaultValue="2"
      onChange={[Function]}
>
      <InlineSelectOption
            active={false}
            onClick={[Function]}
            value="1"
      >
            北京
      </InlineSelectOption>
      <InlineSelectOption
            active={true}
            onClick={[Function]}
            value="2"
      >
            上海
      </InlineSelectOption>
      <InlineSelectOption
            active={false}
            onClick={[Function]}
            value="3"
      >
            南京
      </InlineSelectOption>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <InlineSelect
        defaultValue="2"
        onChange={[Function]}
>
        <InlineSelectOption
                value="1"
        >
                北京
        </InlineSelectOption>
        <InlineSelectOption
                value="2"
        >
                上海
        </InlineSelectOption>
        <InlineSelectOption
                value="3"
        >
                南京
        </InlineSelectOption>
</InlineSelect>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": InlineSelect {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "isPuppet": false,
        "props": Object {
          "children": Array [
            <InlineSelectOption
              value="1"
>
              北京
</InlineSelectOption>,
            <InlineSelectOption
              value="2"
>
              上海
</InlineSelectOption>,
            <InlineSelectOption
              value="3"
>
              南京
</InlineSelectOption>,
          ],
          "defaultValue": "2",
          "onChange": [Function],
        },
        "refs": Object {},
        "state": Object {
          "value": "2",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="inline-select"
          defaultValue="2"
          onChange={[Function]}
>
          <InlineSelectOption
                    active={false}
                    onClick={[Function]}
                    value="1"
          >
                    北京
          </InlineSelectOption>
          <InlineSelectOption
                    active={true}
                    onClick={[Function]}
                    value="2"
          >
                    上海
          </InlineSelectOption>
          <InlineSelectOption
                    active={false}
                    onClick={[Function]}
                    value="3"
          >
                    南京
          </InlineSelectOption>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="inline-select"
          defaultValue="2"
          onChange={[Function]}
>
          <InlineSelectOption
                    active={false}
                    onClick={[Function]}
                    value="1"
          >
                    北京
          </InlineSelectOption>
          <InlineSelectOption
                    active={true}
                    onClick={[Function]}
                    value="2"
          >
                    上海
          </InlineSelectOption>
          <InlineSelectOption
                    active={false}
                    onClick={[Function]}
                    value="3"
          >
                    南京
          </InlineSelectOption>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InlineSelect
    defaultValue="2"
    onChange={[Function]}
>
    <InlineSelectOption
        value="1"
    >
        北京
    </InlineSelectOption>
    <InlineSelectOption
        value="2"
    >
        上海
    </InlineSelectOption>
    <InlineSelectOption
        value="3"
    >
        南京
    </InlineSelectOption>
</InlineSelect>,
}
`;
